lmdeploy/tokenizer.py

```diff
@@ -220,6 +221,42 @@ def __init__(self, model_dir: str):
         self.max_indexes_num = 5
         self.token2id = {}
 
+    def _check_transformers_version(self, model_dir: str):
+        import transformers
+        from packaging import version
+        from transformers import AutoConfig
+
+        from lmdeploy.utils import get_hf_config_content
+
+        logger = get_logger('lmdeploy')
+
+        current_transformers_version = version.parse(transformers.__version__)
+        cfg = get_hf_config_content(model_dir, trust_remote_code=True)
+        required_transformers_version = version.parse(
+            cfg['transformers_version'])
+        if current_transformers_version < required_transformers_version:
+            logger.warning(
+                f'The current version of `transformers` is {current_transformers_version}, '  # noqa: E501
+                f'which is lower than the required version {required_transformers_version}. '  # noqa: E501
+                'Please upgrade to the required version.')
+        try:
+            AutoConfig.from_pretrained(model_dir, trust_remote_code=True)
```

Comment:
AFAK, not all tokenizer configs can be loaded through `AutoConfig`.
```python
from lmdeploy.tokenizer import HuggingFaceTokenizer
tokenizer = HuggingFaceTokenizer('deepseek-ai/deepseek-vl-1.3b-chat')
```

Reply:
ok

Reply:
@AllentDan May I check `model_type` https://huggingface.co/deepseek-ai/deepseek-vl-1.3b-chat/blob/main/config.json#L26 and skip this check when `multi_modality`

Reply:
ERROR - ValueError: The checkpoint you are trying to load has model type `multi_modality` but Transformers does not recognize this architecture. This could be because of an issue with the checkpoint, or because your version of Transformers is out of date.

Reply:
I'm not sure if this is compatible with other models. Do you have a better approach?

Reply:
How about `PretrainedConfig`? All tokenizer config should be able to be loaded by it through `PretrainedConfig.from_pretrained`

Reply:
ok

Reply:
`PretrainedConfig` is ok for `deepseek-vl-1.3b-chat`. But for an error like Qwen 2, it cannot be detected during the check phase.

```python3
from lmdeploy.tokenizer import HuggingFaceTokenizer
tokenizer = HuggingFaceTokenizer('/workdir/deepseek-vl-1.3b-chat')
tokenizer = HuggingFaceTokenizer('/workdir/Qwen2-72B-Instruct')
```

```
2024-07-05 12:02:42,363 - lmdeploy - WARNING - The current version of `transformers` is 4.36.0, which is lower than the required version 4.38.2. Please upgrade to the required version.
The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
You are using a model of type multi_modality to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
Special tokens have been added in the vocabulary, make sure the associated word embeddings are fine-tuned or trained.
2024-07-05 12:02:42,508 - lmdeploy - WARNING - The current version of `transformers` is 4.36.0, which is lower than the required version 4.40.1. Please upgrade to the required version.
The argument `trust_remote_code` is to be used with Auto classes. It has no effect here and is ignored.
You are using a model of type qwen2 to instantiate a model of type . This is not supported for all configurations of models and can yield errors.
Traceback (most recent call last):
  File "/workdir/lmdeploy/test.py", line 3, in <module>
    tokenizer = HuggingFaceTokenizer('/workdir/Qwen2-72B-Instruct')
  File "/workdir/lmdeploy/lmdeploy/tokenizer.py", line 202, in __init__
    self.model = AutoTokenizer.from_pretrained(model_dir,
  File "/usr/local/lib/python3.9/site-packages/transformers/models/auto/tokenization_auto.py", line 784, in from_pretrained
    raise ValueError(
ValueError: Tokenizer class Qwen2Tokenizer does not exist or is not currently imported.
```

Reply:
> AFAK, not all tokenizer configs can be loaded through `AutoConfig`.
> 
> ```python
> from lmdeploy.tokenizer import HuggingFaceTokenizer
> tokenizer = HuggingFaceTokenizer('deepseek-ai/deepseek-vl-1.3b-chat')
> ```

@grimoire How does the PyTorch Engine handle this situation?

Reply:
deepseek-vl has not been supported by pytorch engine yet.

> May I check model_type https://huggingface.co/deepseek-ai/deepseek-vl-1.3b-chat/blob/main/config.json#L26 and skip this check when multimodality

This is acceptable to me. And it would be better if we persuade deepseek-ai to add a `configuration_xxx.py`

Reply:
ok

Reply:
@AllentDan @grimoire I removed the `config load` logic. May you help review the latest code? Thanks.